install_provider.sh

```diff
@@ -0,0 +1,294 @@
+
+        # security headers
+        add_header X-XSS-Protection          "1; mode=block" always;
+        add_header X-Content-Type-Options    "nosniff" always;
+        add_header Referrer-Policy           "no-referrer-when-downgrade" always;
+        add_header Content-Security-Policy   "default-src 'self' http: https: data: blob: 'unsafe-inline'; frame-ancestors 'self';" always;
+        add_header Strict-Transport-Security "max-age=31536000" always;
+
+        auth_basic                           "erigon";
+        auth_basic_user_file                 erigon_htpasswd;
+
+        # reverse proxy
+        location / {
+            proxy_pass                         http://127.0.0.1:8545;
+            proxy_http_version                 1.1;
+            proxy_cache_bypass                 \$http_upgrade;
+
+            # Proxy headers
+            proxy_set_header Upgrade           \$http_upgrade;
+            proxy_set_header Connection        \$connection_upgrade;
+            proxy_set_header Host              \$host;
+            proxy_set_header X-Real-IP         \$remote_addr;
+            proxy_set_header X-Forwarded-For   \$proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto \$scheme;
+            proxy_set_header X-Forwarded-Host  \$host;
+            proxy_set_header X-Forwarded-Port  \$server_port;
+
+            # Proxy timeouts
+            proxy_connect_timeout              60s;
+            proxy_send_timeout                 60s;
+            proxy_read_timeout                 60s;
+        }
+
+        # gzip
+        gzip            on;
+        gzip_vary       on;
+        gzip_proxied    any;
+        gzip_comp_level 6;
+        gzip_types      text/plain text/css text/xml application/json application/javascript application/rss+xml application/atom+xml image/svg+xml;
+    }
+}
+EOF
+
+nginx -t || die "nginx doesn't like it's config"
+systemctl restart nginx
+systemctl enable --now reload-nginx.path
+
+# ========= yagna =========
+
+# install erigon runtime before yagna itself
+mkdir -p "${ERIGON_USER_HOME}/.local/lib/yagna/plugins/ya-runtime-erigon"
+cat >"${ERIGON_USER_HOME}/.local/lib/yagna/plugins/ya-runtime-erigon.json" <<EOF
+[
+  {
+    "name": "erigon",
+    "version": "0.1.0",
+    "supervisor-path": "exe-unit",
+    "runtime-path": "ya-runtime-erigon/ya-erigon-runtime",
+    "description": "Service wrapper for Erigon (formelly Turbo-Geth)",
+    "extra-args": ["--runtime-managed-image"]
+  }
+]
+EOF
+curl -sSfL https://github.com/golemfactory/yagna-service-erigon/releases/download/3deaadc/ya-erigon-runtime.tar.gz \
+    | tar -xzf - -C "${ERIGON_USER_HOME}/.local/lib/yagna/plugins/ya-runtime-erigon" \
+    || die "failed to install erigon runtime"
+
+mkdir -p "${ERIGON_DATADIR}/goerli"
+mkdir -p "${ERIGON_USER_HOME}/.local/share/ya-erigon-runtime"
+
+cat >"${ERIGON_USER_HOME}/.local/share/ya-erigon-runtime/ya-erigon-runtime.json" <<EOF
```

Comment:
This file get out of the script from repo? (see prev comment)

Reply:
I don't think it's justified. It's short, contains a lot of site-specific config and almost no common boilerplate.